plugins/common/opcua/client.go

```diff
@@ -169,6 +169,11 @@ func (o *OpcUAClient) Connect() error {
 	case "opc.tcp":
 		o.State = Connecting
 
+		err = o.Init()
+		if err != nil {
+			return err
+		}
+
```

Comment:
What concerns me a bit here is that a function called `Init()` is called in `Connect()`. While technically speaking this is not an issue it is uncommon for a dev familiar to Telegraf code. Furthermore, this is done conditionally in `opc.tcp` only and thus prone to be forgotten if we ever add other schemes.

Thus my request is to apply one of the following steps:
- if the called `o.Init()` does _not_ make any connections, call `Init()` from the user of the common code in its `Init()` function.
- else remove the `Init()` function (as it is never called from outside?) and call the wrapped `setupOptions()` before the `switch` statement as this is required for _any_ scheme we implement I think.

Reply:
I made these changes and split out the refactor for a future PR! Thanks for the feedback